test/test.js

```diff
@@ -1244,6 +1349,12 @@ describe('Page', function() {
       await button.click('button');
       expect(await page.evaluate(() => result)).toBe('Clicked');
     }));
+    it('should work for TextNodes', SX(async function() {
+      await page.goto(PREFIX + '/input/button.html');
+      const buttonTextNode = await page.evaluateHandle(() => document.querySelector('button').firstChild);
+      await buttonTextNode.click('button');
```

Comment:
element.click doesn't take a string.

This lies about clicking the text node.

Reply:
good call about a string. Regarding clicking text node: i switched to throwing in case of text nodes as we discussed offline.